File: diagscope-test-fixtures/src/main/resources/fixtures/quarkus-entrypoints/src/main/java/example/quarkus/InventoryResource.java
package example.quarkus;

@interface Path { String value() default ""; }
@interface GET { }
@interface POST { }
@interface Scheduled { String every() default ""; }

@Path("/inventory")
class InventoryResource {
    private final InventoryService service = new InventoryService();

    @GET
    @Path("/{id}")
    String read(String id) {
        return service.read(id);
    }

    @POST
    String create() {
        return service.create();
    }

    @Path("/internal")
    String internalOnly() {
        return "not an endpoint";
    }
}

class InventoryService {
    String read(String id) { return id; }
    String create() { return "created"; }
}

class InventoryJobs {
    @Scheduled(every = "15s")
    void refresh() { }
}

File: diagscope-test-fixtures/src/main/resources/fixtures/quarkus-entrypoints/src/main/kotlin/example/quarkus/InventoryResource.kt
package example.quarkus

annotation class Path(val value: String = "")
annotation class GET
annotation class POST
annotation class Scheduled(val every: String = "")

@Path("/inventory")
class InventoryResource(private val service: InventoryService) {
    @GET
    @Path("/{id}")
    fun read(id: String): String = service.read(id)

    @POST
    fun create(): String = service.create()

    @Path("/internal")
    fun internalOnly(): String = "not an endpoint"
}

class InventoryService {
    fun read(id: String): String = id
    fun create(): String = "created"
}

class InventoryJobs {
    @Scheduled(every = "15s")
    fun refresh() { }
}
